api/account/api.go

```diff
@@ -181,23 +181,39 @@ func convertToAccreditations(ctx context.Context, values []string, logger keyclo
 	return &accreds
 }
 
-// ConvertToKCUser creates a KC user representation from an API user
-func ConvertToKCUser(user UpdatableAccountRepresentation) kc.UserRepresentation {
-	var userRep kc.UserRepresentation
+func defaultStringPtr(value *string, defaultValue *string) *string {
+	if value != nil {
+		return value
+	}
+	return defaultValue
+}
 
-	userRep.Username = user.Username
-	userRep.Email = user.Email
-	userRep.FirstName = user.FirstName
-	userRep.LastName = user.LastName
+// MergeUser merge new values into existing KC user representation
+func MergeUser(userRep *kc.UserRepresentation, user UpdatableAccountRepresentation) {
+	userRep.Username = defaultStringPtr(user.Username, userRep.Username)
+	// Don't set email as email and phoneNumber have specific validation rules
```

Comment:
Ok, but where do you manage it now ?

Reply:
It is the responsibility of the caller (UpdateAccount in pkg/account)

Reply:
I do not understand the link between specific validation rules and that we do not merge them ! 
In my point of view a merge method should merge every fields.

Reply:
In my point of view it is really not a good idea !

Reply:
Repushed with renamed function